src/screens/ChannelListScreen/hooks/useFollowUser.ts

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React, {useCallback} from 'react';
+import {useIsFocused} from '@react-navigation/core';
+
+import following from '../../../context/actions/following';
+import {ChannelList as ChannelListData} from '../../../../types/database/schema/ChannelList.types';
+import {Context} from '../../../context';
+import {getFollowing, setFollow, setUnFollow} from '../../../service/profile';
+
+const useFollowUser = () => {
+  const isFocused = useIsFocused();
+  const [followContext, followingDispatch] = (React.useContext(Context) as unknown as any)
+    .following;
+  const [profileContext] = (React.useContext(Context) as unknown as any).profile;
+  const {myProfile} = profileContext;
+
+  const initialFollowingDataRef = React.useRef([...(followContext?.users ?? [])]);
+
+  const updateFollowingData = async () => {
+    try {
+      const response = await getFollowing();
+      following.setFollowingUsers(response.data, followingDispatch);
+    } catch (e) {
+      if (__DEV__) {
+        console.log('error: ', e);
+      }
+      throw new Error(e);
+    }
+  };
+
+  React.useEffect(() => {
+    updateFollowingData();
+  }, [isFocused]);
+
+  const isFollowing = (channel: ChannelListData): boolean => {
+    const targetUser = channel?.rawJson?.channel?.members?.find(
+      (member) => member?.user_id !== myProfile?.user_id
+    )?.user;
+    return Boolean(followContext?.users?.find((user) => user?.user_id_followed === targetUser?.id));
+  };
+
+  const handleFollow = async (channel: ChannelListData) => {
+    const targetUser = channel?.rawJson?.channel?.members?.find(
+      (member) => member?.user_id !== myProfile?.user_id
+    )?.user;
+
+    const data = {
+      user_id_follower: myProfile?.user_id,
+      user_id_followed: targetUser?.id,
+      username_follower: myProfile?.username,
+      username_followed: targetUser?.username ?? targetUser?.name ?? channel?.name,
+      follow_source: 'chat'
+    };
+
+    const isFollowingUser = isFollowing(channel);
+    const payload = {...followContext};
+
+    if (isFollowingUser) {
+      payload?.users?.splice(
+        payload?.users?.findIndex((user) => user?.user_id_followed === targetUser?.id),
+        1
+      );
+    } else {
+      payload?.users?.push({
+        user_id_followed: targetUser?.id,
+        user_id_follower: myProfile?.user_id
+      });
+    }
+
+    following.setFollowingUsers(payload?.users, followingDispatch);
+
+    if (isFollowingUser) {
+      await setUnFollow(data);
+    } else {
+      await setFollow(data);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleFollow` function modifies the `payload.users` array directly by using `splice` and `push`. It's generally not recommended to mutate state directly as it can lead to unexpected behavior. Instead, you should create a new array with the desired changes.
```diff
-  payload?.users?.splice(
-    payload?.users?.findIndex((user) => user?.user_id_followed === targetUser?.id),
-    1
-  );
+  const updatedUsers = payload?.users?.filter(user => user?.user_id_followed !== targetUser?.id);

-  payload?.users?.push({
-    user_id_followed: targetUser?.id,
-    user_id_follower: myProfile?.user_id
-  });
+  const newUsers = [...updatedUsers, {user_id_followed: targetUser?.id, user_id_follower: myProfile?.user_id}];

-  following.setFollowingUsers(payload?.users, followingDispatch);
+  following.setFollowingUsers(isFollowingUser ? updatedUsers : newUsers, followingDispatch);
```

Overall, the hook seems to be well implemented with good use of React hooks and context. However, the improvements suggested above can help make the code more maintainable and easier to understand.


<!-- This is an auto-generated comment by OSS CodeRabbit -->